tests/integration/pilot/istioctl_test.go

```diff
@@ -172,3 +180,45 @@ func getPodID(i echo.Instance) (string, error) {
 
 	return "", fmt.Errorf("no workloads")
 }
+
+func TestAddToAndRemoveFromMesh(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		RunParallel(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "istioctl-add-to-mesh",
+				Inject: true,
+			})
+
+			var a echo.Instance
+			echoboot.NewBuilderOrFail(ctx, ctx).
+				With(&a, echoConfig(ns, "a")).
+				BuildOrFail(ctx)
+
+			if err := a.WaitUntilCallable(a); err != nil {
+				t.Fatal(err)
+			}
+			istioCtl := istioctl.NewOrFail(t, ctx, istioctl.Config{})
+
+			var output string
+			var args []string
+			g := gomega.NewGomegaWithT(t)
+
+			// able to remove from mesh when the deployment is auto injected
+			args = []string{fmt.Sprintf("--namespace=%s", ns.Name()),
+				"x", "remove-from-mesh", "service", "a"}
+			output = istioCtl.InvokeOrFail(t, args)
+			g.Expect(output).To(gomega.MatchRegexp(removeFromMeshPodAOutput))
+
+			// remove from mesh should be clean
+			// users can add it back to mesh successfully
+			if err := a.WaitUntilCallable(a); err != nil {
```

Comment:
Was this what fixed the flakes?

Reply:
I think so - there is definitely some timing related issue so wanted to make sure the service is up running.  Also, it seems describe test failed sometimes, which caused the following next to fail so I fixed the describe test also.